src/overlays/actors/ovl_Boss_Ganon/z_boss_ganon.c

```diff
@@ -108,127 +98,4928 @@ static ColliderCylinderInit D_808E4C2C = {
+                    hitWithBottle = true;
+                } else {
+                    hitWithBottle = false;
+                }
+
+                if ((this->collider.base.acFlags & 2) || hitWithBottle) {
+                    ColliderInfo* acHitInfo = this->collider.info.acHitInfo;
+                    this->collider.base.acFlags &= ~2;
+
+                    if ((hitWithBottle == false) && (acHitInfo->toucher.dmgFlags & 0x100000)) {
+                        spBA = 2;
+                        Audio_PlaySoundGeneral(NA_SE_IT_SHIELD_REFLECT_MG, &player->actor.projectedPos, 4, &D_801333E0,
+                                               &D_801333E0, &D_801333E8);
+                        func_800AA000(this->actor.xyzDistToPlayerSq, 0xFF, 0x14, 0x96);
+                    } else {
+                        spBA = 1;
+                        this->actor.world.rot.y = Math_Atan2S(zDistFromGanondorf, xDistFromGanondorf);
+                        this->actor.world.rot.x =
+                            Math_Atan2S(sqrtf(SQ(xDistFromGanondorf) + SQ(zDistFromGanondorf)), yDistFromGanondorf);
+                        this->unk_1A4++;
+                        this->timers[1] = 2;
+                        Audio_PlaySoundGeneral(NA_SE_IT_SWORD_REFLECT_MG, &player->actor.projectedPos, 4, &D_801333E0,
+                                               &D_801333E0, &D_801333E8);
+                        func_800AA000(this->actor.xyzDistToPlayerSq, 0xB4, 0x14, 0x64);
+
+                        if (hitWithBottle == false) {
+                            // if ganondorf is 250 units away from link, at least 3 volleys are required
+                            if ((ganondorf->actor.xyzDistToPlayerSq > 62500.0f) && (this->unk_1A4 < 3)) {
+                                this->unk_1C2 = 1;
+                            } else if (Rand_ZeroOne() < 0.7f) {
+                                this->unk_1C2 = 1;
+                            } else {
+                                this->unk_1C2 = 3;
+                            }
+
+                            // if a spin attack is used
+                            if (player->swordAnimation >= 0x18) {
+                                this->actor.speedXZ = 20.0f;
+                            }
+                            break;
+                        } else {
+                            if (Rand_ZeroOne() < 0.9f) {
+                                this->unk_1C2 = 1;
+                            } else {
+                                this->unk_1C2 = 3;
+                            }
+                        }
+                    }
+                } else {
+                    if (sqrtf(SQ(xDistFromLink) + SQ(yDistFromLink) + SQ(zDistFromLink)) <= 25.0f) {
+                        spBA = 5;
+                        func_8002F6D4(globalCtx, &this->actor, 3.0f, this->actor.world.rot.y, 0.0f, 0x30);
+                        Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 0x28, NA_SE_EN_GANON_HIT_THUNDER);
```

Comment:
```suggestion
                        Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 40, NA_SE_EN_GANON_HIT_THUNDER);
```
These are duration/timers, so should probably be in dec